constraint/pkg/client/template_client.go

```diff
@@ -59,7 +60,22 @@ func (e *templateClient) ValidateConstraint(constraint *unstructured.Unstructure
 		}
 	}
 
-	return crds.ValidateCR(constraint, e.crd)
+	if err := crds.ValidateCR(constraint, e.crd); err != nil {
+		return err
+	}
+
+	// validate that there are no unknown fields in the CR
+	// NewStructural assumes the schema is validate
+	structural, err := schema.NewStructural(e.crd.Spec.Validation.OpenAPIV3Schema)
```

Comment:
This should go into the ValidateCR function.

Reply:
Moved to `ValidateCR`